lua/lspconfig/server_configurations/grammarly.lua

```diff
@@ -0,0 +1,33 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    filetypes = { 'markdown' },
+    root_dir = util.find_git_ancestor,
+  },
+  docs = {
+    description = [[
+Grammarly in an LSP, using the Grammarly API.
+
+Requires some manual installation:
+
+1. Clone the Grammarly repo, the upstream is [here](https://github.com/znck/grammarly), it's designed to work with a custom client though, so it requires some modifications. I'd recommend using [this branch of my fork](https://github.com/mtoohey31/grammarly/tree/generic-client-compat) might work for you.
+2. Build the server:
+
+```bash
+pnpm run build:packages && pnpm run build:server
+```
+
+(you can install pnpm with `npm install -g pnpm`)
```

Comment:
Why do you need pnpm?

Reply:
The original project uses pnpm, so the lock file in the repo is a `pnpm-lock.yaml`. If you install things with npm, there are dependency issues and the project fails to build, though I suppose pnpm is really only needed for the install command, which is the one I forgot to metion... I'll fix that!

Reply:
Fixed by 63bb6276189a98c942fc7cb77300634e2a3cb5ce.